spec/integration/apply_spec.rb

```diff
@@ -38,6 +59,55 @@ def root_config
       run_command(uninstall, uri, config: config_data, inventory: inventory_data)
     end
 
+    context "when running against puppet 5 or puppet 6" do
+      it 'installs puppet agent' do
+        # install puppet5
+        result = run_task('puppet_agent::install', 'puppet_5', { 'collection' => 'puppet5' },
+                          config: root_config, inventory: agent_version_inventory)
+        expect(result.count).to eq(1)
+        expect(result[0]['status']).to eq('success')
+
+        result = run_task('puppet_agent::version', 'puppet_5', inventory: agent_version_inventory)
+        expect(result.count).to eq(1)
+        expect(result[0]['status']).to eq('success')
+        expect(result[0]['result']['version']).to match(/^5/)
+
+        # install puppet6
+        result = run_task('puppet_agent::install', 'puppet_6', { 'collection' => 'puppet6' },
+                          config: root_config, inventory: agent_version_inventory)
+        expect(result.count).to eq(1)
+        expect(result[0]['status']).to eq('success')
+
+        result = run_task('puppet_agent::version', 'puppet_6', inventory: agent_version_inventory)
+        expect(result.count).to eq(1)
+        expect(result[0]['status']).to eq('success')
+        expect(result[0]['result']['version']).to match(/^6/)
+      end
+
+      it 'runs a ruby task' do
+        with_tempfile_containing('inventory', YAML.dump(agent_version_inventory), '.yaml') do |inv|
+          results = run_cli_json(%W[task run basic::ruby_task --nodes agent_targets
+                                    --modulepath #{modulepath} --inventoryfile #{inv.path}])
+          results['items'].each do |result|
+            expect(result['status']).to eq('success')
+            expect(result['result']).to eq('ruby' => 'Hi')
+          end
+        end
+      end
+
+      it 'runs an apply plan' do
+        with_tempfile_containing('inventory', YAML.dump(agent_version_inventory), '.yaml') do |inv|
+          results = run_cli_json(%W[plan run basic::notify --nodes agent_targets
+                                    --modulepath #{modulepath} --inventoryfile #{inv.path}])
+          results.each do |result|
```

Comment:
This was passing earier today on travis and was passing locally. In order to reproduce locally I rebased to upstream/master and destroyed my Gemfile.lock/re-installed. 

The error message (for both puppet5 and 6) for the provided task is: 
```
      "result"=>
       {"_error"=>
         {"kind"=>"puppetlabs.tasks/task-error",
          "issue_code"=>"TASK_ERROR",
          "msg"=>
           "The task failed with exit code 1:\n/opt/puppetlabs/puppet/lib/ruby/v
endor_ruby/puppet/configurer.rb:54:in `rescue in rescue in init_storage': Cannot
 remove /tmp/d20181003-792-xynxz5/state/state.yaml: No such file or directory @ 
unlink_internal - /tmp/d20181003-792-xynxz5/state/state.yaml (Puppet::Error)\n\t
from /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/configurer.rb:50:in `res
cue in init_storage'\n\tfrom /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/
configurer.rb:45:in `init_storage'\n\tfrom /opt/puppetlabs/puppet/lib/ruby/vendo
r_ruby/puppet/configurer.rb:205:in `run'\n\tfrom /tmp/09555e1f-b723-4189-8264-d4
1c049c312a/apply_catalog.rb:81:in `block in <main>'\n\tfrom /opt/puppetlabs/pupp
et/lib/ruby/vendor_ruby/puppet/context.rb:65:in `override'\n\tfrom /opt/puppetla
bs/puppet/lib/ruby/vendor_ruby/puppet.rb:260:in `override'\n\tfrom /tmp/09555e1f-b723-4189-8264-d41c049c312a/apply_catalog.rb:69:in `<main>'\n",
          "details"=>{"exit_code"=>1}}}},
```

Reply:
Issue appears to only arises when running apply plan as non-root user.